paymentsheet/src/androidTest/java/com/stripe/android/paymentsheet/LinkTest.kt

```diff
@@ -0,0 +1,445 @@
+
+    private val page: PaymentSheetPage = PaymentSheetPage(composeTestRule)
+
+    @TestParameter(valuesProvider = LinkIntegrationTypeProvider::class)
+    lateinit var integrationType: LinkIntegrationType
+
+    @Test
+    fun testSuccessfulCardPaymentWithLinkSignUp() = activityScenarioRule.runLinkTest(
+        integrationType = integrationType,
+        paymentOptionCallback = { paymentOption ->
+            assertThat(paymentOption?.label).endsWith("4242")
+        },
+        resultCallback = ::assertCompleted,
+    ) { testContext ->
+        networkRule.enqueue(
+            host("api.stripe.com"),
+            method("GET"),
+            path("/v1/elements/sessions"),
+        ) { response ->
+            response.testBodyFromFile("elements-sessions-requires_payment_method.json")
+        }
+
+        testContext.launch()
+
+        page.fillOutCardDetails()
+
+        networkRule.enqueue(
+            method("POST"),
+            path("/v1/consumers/sessions/lookup"),
+        ) { response ->
+            response.testBodyFromFile("consumer-session-lookup-success.json")
+        }
+
+        page.clickOnLinkCheckbox()
+        page.fillOutLinkEmail()
+        page.fillOutLinkPhone()
+
+        Espresso.closeSoftKeyboard()
+
+        networkRule.enqueue(
+            method("POST"),
+            path("/v1/consumers/accounts/sign_up"),
+        ) { response ->
+            response.testBodyFromFile("consumer-accounts-signup-success.json")
+        }
+
+        networkRule.enqueue(
+            method("POST"),
+            path("/v1/consumers/payment_details"),
+            /*
+             * Ensures we are passing the full expiration year and not the
+             * 2-digit shorthand (should send "2034", not "34")
+             */
+            bodyPart("card%5Bexp_year%5D", "2034"),
+            /*
+             * Should use the consumer's publishable key when creating payment details
+             */
+            header("Authorization", "Bearer pk_545454676767898989"),
+        ) { response ->
+            response.testBodyFromFile("consumer-payment-details-success.json")
+        }
+
+        networkRule.enqueue(
+            method("POST"),
+            path("/v1/payment_intents/pi_example/confirm"),
+            linkInformation()
+        ) { response ->
+            response.testBodyFromFile("payment-intent-confirm.json")
+        }
+
+        networkRule.enqueue(
+            method("POST"),
+            path("/v1/consumers/sessions/log_out"),
+        ) { response ->
+            response.testBodyFromFile("consumer-session-logout-success.json")
+        }
+
+        page.clickPrimaryButton()
+    }
+
+    @Test
+    fun testSuccessfulCardPaymentWithLinkSignUpAndLinkPassthroughMode() = activityScenarioRule.runLinkTest(
+        integrationType = integrationType,
+        paymentOptionCallback = {
+            // TODO(samer-stripe): Re-enable this assertion once unknown payment option bug is fixed
```

Comment:
This came up in testing. The returned payment option is broken in passthrough mode with `FlowController`. There is no `Link` logo shown or last 4 digits of the card/bank account used.